### DISCUSSION TITLE
Non-technical way to block tracking for internal traffic

### DISCUSSION THREAD

**Discussion by LushawnDev** [umami-software/umami#2888] [Ideas]
Firstly, I love Umami. You've created my favourite analytics software by far!

At the moment the only way to block tracking for internal traffic is by adding an `umami.disabled` to `true` in local storage. You can do this manually or using a quick line in the browser's console.

This is great for technical users and developers like myself, but for non-technical people, it's a pretty foreign task.

When I set clients up with Umami, it'd be great to have a less technical way for them to block tracking for their own browsing of their site because they end up getting such skewed data when they start checking out their own site.

How exactly you could do this, I'm not too sure. I assume IP address blocking isn't possible due to Umami's privacy-friendly nature.

Perhaps a Chrome extension that can set that in your local storage for you?

Maybe even an official page on the Umami website where you can go to set something in their local storage for their own site? Although that could be tricky setting cross-domain local storage, [here is one idea how it could be done](https://adiachituve.medium.com/how-to-achieve-cross-domain-localstorage-790a657ec36f).

Or the minimum would be some nice clear docs on the Umami website which non-technical users could be referred to and get walked through the process more clearly with some screenshots, etc.

Thanks, and keen to hear others' thoughts! (or perhaps there is an easy way to do this that I'm unaware of)

**Comment by LushawnDev** [comment]
I was happy to see the improvement you made in the latest release. Just bumping this to see if there's any possibility of this? I've currently got a client with really lopsided analytics from their internal data haha.

**Comment by ethanator** [comment]
I wanted to echo the sentiment — our organization has also encountered this issue, and it’s been a significant pain point for our analytics. We’d greatly appreciate it if the Umami team could prioritize implementing this feature.

**Comment by jaroslaw-weber** [comment]
how about just adding a "secret" route that user can open that will add the value to local storage?

**Reply by mikecao** [reply]
This seems like the best option.

**Comment by jaroslaw-weber** [comment]
also can detect if user is you by disabling tracking on localhost or when u login

```
export function trackingSetup() {
	if (window) {
		if (window.location.hostname === "localhost") {
			disableTracking("local development");
		}

	}
}
export function disableTracking(reason:string) {
	console.log("disabling Umami tracking: " + reason);
	localStorage.setItem("umami.disabled", "true");
}

```
